usr/src/common/zfs/zpool_prop.c

```diff
@@ -136,6 +136,8 @@ zpool_prop_init(void)
 	    PROP_READONLY, ZFS_TYPE_POOL, "NAME");
 	zprop_register_hidden(ZPOOL_PROP_MAXBLOCKSIZE, "maxblocksize",
 	    PROP_TYPE_NUMBER, PROP_READONLY, ZFS_TYPE_POOL, "MAXBLOCKSIZE");
+	zprop_register_hidden(ZPOOL_PROP_TNAME, "tname", PROP_TYPE_STRING,
+	    PROP_ONETIME, ZFS_TYPE_POOL, "TNAME");
```

Comment:
This should really be `"tempname"` and `ZPOOL_PROP_TEMPNAME`.  Unfortunately, ZoL has already set the precedent of `-o tname=`, so I guess we should stick with that.